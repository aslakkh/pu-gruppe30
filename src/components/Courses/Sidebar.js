import React, { Component } from 'react'
import {Tab,NavItem, Nav, Col,Row} from 'react-bootstrap'
import SpecificCourse from './specificCourse'

/*Maps the courses to make the sidebar and the site for the specific courses */
function TestEmner(props){

	let course = Object.keys(props.courses).map((emne,i) =>
        
        <NavItem eventKey={emne} className="navbar-link" key={i} >
            {emne}
        </NavItem>
        

			)
	let info = Object.keys(props.courses).map((emne,i) =>
		<Tab.Pane eventKey={emne} key={i}>
            <h1>{emne}</h1>
            <SpecificCourse courseID={emne} course={props.courses[emne]}/>
    </Tab.Pane>
 
		)

	return(

	<Tab.Container id="left-tabs" defaultActiveKey={0} style={{border: 'white', marginleft: '0px'}}>
    <Row className="clearfix">
      <Col sm={3}>
        <Nav bsStyle="pills" className="navbar navbar-inverse" stacked>
        {course}
        </Nav>
      </Col>
            <Col sm={9}>
        <Tab.Content animation>
        	{info}
        </Tab.Content>
      </Col>
    </Row>
  </Tab.Container>





		)


};



export default class Courses extends Component {
constructor(props) {
  super(props);
  this.state={
			
			courses: this.props.courses
		}
  
};

componentWillReceiveProps(nextProps){
  this.setState({
    courses: nextProps.courses
  });
}

componentWillMount(){
  /*const that = this;
    const userUid = firebase.auth().currentUser.uid;
    ref.child('users/'+userUid+'/courses').once("value",function(snapshot){
      snapshot.forEach(function(data){
        that.emner.push(data.key)
        that.setState({})
    })})*/

}
	render (){

    if(this.state.courses === undefined || this.state.courses === null){ //checks if props are loaded
      return(
        <h4>ingenting</h4>
      );
    }
    else{
      return (
		    <TestEmner courses={this.state.courses}/>
			)
    }
		
	}




}