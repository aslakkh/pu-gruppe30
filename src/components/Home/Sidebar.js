import React, { Component } from 'react'
import {Tab,NavItem, Nav, Col,Row} from 'react-bootstrap'
import SpecificCourse from './specificCourse'

/*Maps the courses to make the sidebar and the site for the specific courses
*
*
* Changes:
* Change the way SpecificCourse is called, so you only load the course you are viewing.
*
* */




export default class Courses extends Component {
constructor(props) {
  super(props);
  this.state={
			
			courses: this.props.courses
		}
  
};

componentWillReceiveProps(nextProps){
  this.setState({
    courses: nextProps.courses
  });
}
    handleSelect(key){
        this.setState({
            key:key

        })

    }



componentWillMount(){
}
	render (){

    if(this.state.courses === undefined){ //checks if props are loaded
      return(
        <h4>ingenting</h4>
      );
    }
    else{
      return (

            <Tab.Container id="left-tabs" defaultActiveKey={(Object.keys(this.state.courses)[0])} activeKey={this.state.key}>
                <Row className="clearfix">
                <Col sm={3}>
                <Nav bsStyle="pills" className="navbar navbar-inverse" stacked>
                    {Object.keys(this.state.courses).map((emne,i) =>

                    <NavItem eventKey={emne} className="navbar-link" key={i} onSelect={this.setState({key:emne})} >
                        {emne}
                    </NavItem>


                    )}
                </Nav>
                </Col>
                <Col sm={9}>
                <Tab.Content animation>
                    {Object.keys(this.state.courses).map((emne,i) =>
                    <Tab.Pane eventKey={emne} key={i}>
                        <h1>{emne}</h1>
                        <SpecificCourse courseID={emne} course={this.state.courses[emne]}/>
                    </Tab.Pane>

                    )}
                </Tab.Content>
                </Col>
                </Row>
                </Tab.Container>









			)
    }
		
	}




}