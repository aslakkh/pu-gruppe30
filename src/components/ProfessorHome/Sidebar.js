import React, { Component } from 'react'
import {Tab,NavItem, Nav, Col,Row} from 'react-bootstrap'
import AddMessage from './AddMessage'
import MessageList from './MessageList'
import Statistics from './Statistics'
import SetTime from './SetTime'
import {
    BrowserRouter as Router,
    Route,
    Link,
    NavLink
} from 'react-router-dom'
import {LinkContainer} from 'react-router-bootstrap'

/*Maps the courses to make the sidebar and the site for the specific courses
 *
 *
 * Changes:
 * Change the way SpecificCourse is called, so you only load the course you are viewing.
 *
 * */

export default class Courses extends Component {
constructor(props) {
  super(props);
  this.state={
      courses: this.props.courses,
		}
  this.handleRoute = this.handleRoute.bind(this)
};

componentWillReceiveProps(nextProps){
  this.setState({
    courses: nextProps.courses
  });
}
    //matches URl (match.params.id) to course
    handleRoute({match}){
        return(
            <div>
                <AddMessage courseID={match.params.id}/>
                <MessageList courseID={match.params.id}/>
                <SetTime courseID={match.params.id} />
                <Statistics courseID={match.params.id}/>
            </div>
        )
    }
    
componentWillMount(){
}
	render (){

    if(this.state.courses === undefined || this.state.courses === null){ //checks if props are loaded
      return(
        <h4>No courses</h4>
      );
    }
    else{
      return (
          <Router>
              <div>
                  <Tab.Container id="left-tabs" >
                      <Row className="clearfix">
                          <Col sm={3}>
                              <Nav bsStyle="pills" className="navbar navbar-inverse" stacked >
                                  {Object.keys(this.state.courses).map((emne,i) =>


                                          <NavItem key={i} eventKey={i} className="navbar-link">
                                            <Link to={"/"+emne} key={i}>
                                                <div style={{marginTop: '-0.3em', fontSize:'1.5em',marginBottom: '-0.3em', color:'darkgrey'}}>
                                                {emne}
                                                </div>
                                            </Link>
                                    </NavItem>
                                  )}
                              </Nav>
                          </Col>
                          <Col sm={9}>
                              <Tab.Content animation>
                                  <Route path="/:id" render={this.handleRoute} />
                              </Tab.Content>
                          </Col>
                      </Row>
                  </Tab.Container>
              </div>
          </Router>
      )
    }
		
	}




}