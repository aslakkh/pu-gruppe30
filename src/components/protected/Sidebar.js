import React, { Component } from 'react'
import Course from './Course'
import {Tab,NavItem, Nav, Col,Row} from 'react-bootstrap'
import { ref } from '../../config/constants'
import firebase from 'firebase';
import getCourse from '../index'
import { BrowserRouter as Router, Route, Switch, Link, Redirect } from 'react-router-dom'


function TestEmner(props){
	let emner = props.emner;

	let course = emner.map((emne,i) =>
        <NavItem  eventKey={i} key={i} >
            <div  className="navbar-link">{emne}</div>
        </NavItem>

			)
	let info = emner.map((emne,i) =>
		<Tab.Pane eventKey={i} key={i}>
            {emne} Stuff goes here
          </Tab.Pane>
 
		)

	return(

	<Tab.Container id="left-tabs-example" defaultActiveKey={0} style={{border: 'black', marginleft: '0px'}}>
    <Row className="clearfix">
      <Col sm={2}>
        <Nav bsStyle="pills" className="navbar navbar-inverse" stacked>
        {course}
        </Nav>
      </Col>
            <Col sm={10}>
        <Tab.Content animation>
        	{info}
        </Tab.Content>
      </Col>
    </Row>
  </Tab.Container>





		)


};



export default class Courses extends Component {
constructor(props) {
  super(props);
  this.emner=[]
  this.state=({emner:["hehe"]})
};


componentWillMount(){
  const that = this;
    const userUid = firebase.auth().currentUser.uid;
    ref.child('users/'+userUid+'/courses').once("value",function(snapshot){
      snapshot.forEach(function(data){
        console.log(data.val(),data.key)
        that.emner.push(data.key)
        that.setState({})
    })})

}
	render (){
		return (


		<TestEmner emner ={this.emner}/>




			)
	}




}