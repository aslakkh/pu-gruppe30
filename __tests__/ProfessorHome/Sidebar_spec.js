/**
 * Created by jan on 19/04/2017.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import Sidebar from '../../src/components/ProfessorHome/Sidebar.js';
import TestUtils from "react-addons-test-utils"
const courses={
    TES1000:{active: true,
        goals: {dailyGoal:{active: false,
            timeSet: 0,
            timeSpent: 0,
            value: 0},
            monthlyGoal:
                {active:false,
                    timeSet:0,
                    timeSpent:0,
                    value:0},
            weeklyGoal:
                {active:false,
                    timeSet:0,
                    timeSpent:0,
                    value:0}

        }
    }};
const courseNull = undefined;
describe("Sidebar", function () {
    let renderer2;
    let renderer3;
    let renderer4;
    let component2;
    let component3;
    let component4;
    renderer2 = TestUtils.createRenderer();
    renderer2.render(
        <Sidebar courses={courseNull}/>
    );
    component2 = renderer2.getRenderOutput();
    it("No courses", () => {
        expect(component2).toMatchSnapshot();
    });
    //renderer3 = TestUtils.createRenderer();
    renderer2.render(
        <Sidebar courses={courses}/>
    );
    component2 = renderer2.getRenderOutput();
    it("One course", () => {
        expect(component2).toMatchSnapshot();
    });
});