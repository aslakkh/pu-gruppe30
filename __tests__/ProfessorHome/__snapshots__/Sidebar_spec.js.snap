// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar No courses 1`] = `
<BrowserRouter>
  <div>
    <Uncontrolled(TabContainer)
      id="left-tabs"
    >
      <Row
        bsClass="row"
        className="clearfix"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          sm={3}
        >
          <div
            className="navbar navbar-inverse nav-stacked"
          >
            <div>
              <Link
                replace={false}
                to="TES1000"
              >
                TES1000
              </Link>
            </div>
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          sm={9}
        >
          <TabContent
            animation={true}
            bsClass="tab"
            componentClass="div"
            mountOnEnter={false}
            unmountOnExit={false}
          >
            <Route
              path="/:id"
              render={[Function]}
            />
          </TabContent>
        </Col>
      </Row>
    </Uncontrolled(TabContainer)>
  </div>
</BrowserRouter>
`;

exports[`Sidebar One course 1`] = `
<BrowserRouter>
  <div>
    <Uncontrolled(TabContainer)
      id="left-tabs"
    >
      <Row
        bsClass="row"
        className="clearfix"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          sm={3}
        >
          <div
            className="navbar navbar-inverse nav-stacked"
          >
            <div>
              <Link
                replace={false}
                to="TES1000"
              >
                TES1000
              </Link>
            </div>
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          sm={9}
        >
          <TabContent
            animation={true}
            bsClass="tab"
            componentClass="div"
            mountOnEnter={false}
            unmountOnExit={false}
          >
            <Route
              path="/:id"
              render={[Function]}
            />
          </TabContent>
        </Col>
      </Row>
    </Uncontrolled(TabContainer)>
  </div>
</BrowserRouter>
`;
